packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/service/NotificationResponseReceiver.java

```diff
@@ -34,16 +38,51 @@ public static PendingIntent getActionIntent(Context context, String actionIdenti
 
   @Override
   public void onReceive(Context context, Intent intent) {
-    openAppToForeground(context);
-    BaseNotificationsService.enqueueResponseReceived(context, intent.<NotificationResponse>getParcelableExtra(NOTIFICATION_RESPONSE_KEY));
+    NotificationResponse response = intent.getParcelableExtra(NOTIFICATION_RESPONSE_KEY);
+    openAppToForeground(context, response);
+    BaseNotificationsService.enqueueResponseReceived(context, response);
   }
 
-  protected void openAppToForeground(Context context) {
+  protected void openAppToForeground(Context context, NotificationResponse notificationResponse) {
+    Intent notificationAppHandler = new Intent(NOTIFICATION_OPEN_APP_ACTION);
```

Comment:
Do you think it would be worth it to extract these two different ways of fetching activity to two separate cleanly named functions?